fvm/scriptEnv.go

```diff
@@ -89,9 +91,33 @@ func NewScriptEnvironment(
 		env.seedRNG(ctx.BlockHeader)
 	}
 
+	env.setMeteringWeights()
```

Comment:
Does this get called for every script/tx?

Reply:
in general once per script/transaction yes.

When it comes to transactions, the value being read from storage will already be cached the second time it is needed in a collection.

Reply:
I see! Thanks for the explanation. My concern was on having to encode/decode the entire map every time for each tx. Not a problem for this PR, I'm just thinking out loud whether it's possible to optimize that sometime in future 👌